hazelcast/src/main/java/com/hazelcast/jet/impl/JobRepository.java

```diff
@@ -519,21 +521,28 @@ private boolean isResourceMapExpired(long creationTime) {
 
     Set<Long> getAllJobIds() {
         Set<Long> ids = new HashSet<>();
-        ids.addAll(jobRecords.keySet());
-        ids.addAll(jobResults.keySet());
+        ids.addAll(jobRecords.get().keySet());
+        ids.addAll(jobResults.get().keySet());
         return ids;
     }
 
     public Collection<JobRecord> getJobRecords() {
-        return jobRecords.values();
+        return jobRecords.get().values();
+    }
+
+    public boolean jobRecordsExists() {
+        if (jobRecords.remembered() != null) {
+            return true;
+        }
+        return instance.getDistributedObjects().stream().anyMatch(o -> o.getName().equals(JOB_RECORDS_MAP_NAME));
```

Comment:
From `getDistributedObjects` javadoc:

>The result might miss just-created objects and contain just-deleted objects. An existing object can also be missing from the list occasionally.

This code can make bad situations worse. It certainly won't be tested for this scenario and if there's some network hiccup and this member becomes a master, it might not run jobs simply because it incorrectly thought the `JobRecords` map doesn't exist...

Do we really need to avoid the "existence" of jet maps even for such cost? Can't we just say "well, there's such and such map and it's internal, ignore it". Hazelcast doesn't have proper object lifecycle, we shouldn't workaround it like this IMO.

There's also `AbstractJetInstance.existsDistributedObject()` method that might be useful.

Reply:
I've reverted the scanning logic changes instead, now I check if the internal map exists.